cli/CampaignConfigValidation/LoggingCampaignConfigurationLoader.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation;
+
+use Symfony\Component\Yaml\Exception\ParseException;
+use WMDE\Fundraising\Frontend\BucketTesting\CampaignConfigurationLoaderInterface;
+
+class LoggingCampaignConfigurationLoader implements CampaignConfigurationLoaderInterface {
+
+	private $actualCampaignConfigurationLoader;
+	private $errorLogger;
+
+	public function __construct( CampaignConfigurationLoaderInterface $campaignConfigurationLoader, ValidationErrorLogger $errorLogger ) {
+		$this->errorLogger = $errorLogger;
+		$this->actualCampaignConfigurationLoader = $campaignConfigurationLoader;
+	}
+
+	public function loadCampaignConfiguration( string ...$configFiles ): array {
+		try {
+			return $this->actualCampaignConfigurationLoader->loadCampaignConfiguration( ...$configFiles );
+		}
+		catch ( ParseException $e ) {
+			$this->errorLogger->addError( 'Failed to parse YAML: ' . $e->getMessage() );
+		}
+		catch ( \LogicException $e ) {
```

Comment:
I don't remember `LogicException` being thrown ...

Reply:
See `\WMDE\Fundraising\Frontend\BucketTesting\Campaign::getDefaultBucket()`.

Reply:
Yes, but that's not called when loading the configuration, only during testing and when buckets are assigned. If a LogicException occurs, I think it's better to lump it with the handling of the other unexpected exceptions.

Reply:
Ah alright, thanks for pointing that out.